### PR TITLE
Allow custom clock function

### PR DESCRIPTION
**Pull request by alindeman**
This will make Verifier more easy to use in tests.

### CODE REVIEW

File: verifier.go [hunk 1]
```diff
@@ -35,9 +38,15 @@ func NewVerifier(issuer string, keySource KeySource) *Verifier {
 	}
 }
 
-type verifyCfg struct{}
+type VerifyOpt func(*Verifier)
```
Comment (alindeman) [inline verifier.go:41]: @lstoll Do you think we need to keep `verifyCfg`? I couldn't really figure out a good reason to use it here. Our `ClientOpt` concept in `client.go` also just takes a `*Client`.

If you think there is a better way to accomplish this, though, let me know.

File: verifier.go [hunk 2]
```diff
@@ -35,9 +38,15 @@ func NewVerifier(issuer string, keySource KeySource) *Verifier {
...
-type VerifyOpt func(v *verifyCfg)
+// WithClock provides a custom clock that is used to determine the current time
+// when verifying tokens.
+func WithClock(clock func() time.Time) VerifyOpt {
+	return func(v *Verifier) {
+		v.clock = clock
+	}
+}
 
 func (v *Verifier) VerifyRaw(ctx context.Context, audience string, raw string, opts ...VerifyOpt) (*Claims, error) {
```
Comment (lstoll) [inline verifier.go:51]: nothing in this function appears to be calling the opts on the verifier, so might need to add something

File: verifier.go (hunk 1 above)
Comment (alindeman) [reply verifier.go:41]: This is a great point that I didn't consider. I'm going to change it back to using `verifyCfg` that is local to the `VerifyRaw` call.

Thank you for making me reconsider.

**Review by lstoll — APPROVED** [review]

**Comment by alindeman** [comment]
This ended up not being needed.

**Pull request closed without merging**